### ISSUE TITLE
'x' arg in prior trainer

### ISSUE THREAD

**Issue by nousr** [lucidrains/DALLE2-pytorch#98]
https://github.com/lucidrains/DALLE2-pytorch/blob/23bb206d2710392a99c879c37cd7ec71665ddac7/dalle2_pytorch/train.py#L259-L265

what is 'x' here? I tried calling like the updated way in train_diffusion_prior.py (except I do text=.., img_embed=...)
https://github.com/lucidrains/DALLE2-pytorch/blob/23bb206d2710392a99c879c37cd7ec71665ddac7/train_diffusion_prior.py#L218
and I'm getting an error `forward() missing 1 required positional argument: 'x'`

**Comment by lucidrains** [comment]
@nousr oh hey! fancy seeing you working on a sunday lol

yea, so i was trying to make gradient accumulation automatically work for researchers with not enough memory. i was going to build in the assumption that the tensor with the batch element on dimension 0 is given as the first argument, but realized that isn't a good assumption

let me know if https://github.com/lucidrains/DALLE2-pytorch/commit/68e7d2f2419822676afe0b26915b1aa0e53bd248 works!

**Comment by lucidrains** [comment]
@nousr hey! so i was trying to make gradient accumulation automatically work for researchers with not enough memory. i was going to build in the assumption that the tensor with the batch element on dimension 0 is given as the first argument, but realized that isn't a good assumption

let me know if https://github.com/lucidrains/DALLE2-pytorch/commit/68e7d2f2419822676afe0b26915b1aa0e53bd248 fixes things

**Comment by nousr** [comment]
> let me know if https://github.com/lucidrains/DALLE2-pytorch/commit/68e7d2f2419822676afe0b26915b1aa0e53bd248 fixes things

ahh! okay okay, i see

i'm at the car wash rn, will try when i get home 🤘

**Comment by lucidrains** [comment]
Lol ok, yeah was surprised to see you already trying my changes on the weekend

**Comment by nousr** [comment]
> Lol ok, yeah was surprised to see you already trying my changes on the weekend

haha yeah, I finished some assignments early last week so I had time to mess around :)

& the code doesn't seem to yell at me anymore after https://github.com/lucidrains/DALLE2-pytorch/commit/68e7d2f2419822676afe0b26915b1aa0e53bd248 👍 thanks!

**Comment by lucidrains** [comment]
@nousr noice :^)
